packages/kbn-optimizer/limits.yml

```diff
@@ -20,7 +20,7 @@ pageLoadAssetSize:
   controls: 40000
   core: 435325
   crossClusterReplication: 65408
-  customIntegrations: 28810
+  customIntegrations: 45173
```

Comment:
FYI, this is still relatively small but the changes you've made here really shouldn't need to be loaded at page load for every application in Kibana.

I took a look at how this plugin works though, and it seems like the whole purpose of this plugin is to encapsulate two components and a context provider where are each exposed on the start contract, meaning this can't really be async loaded without turning each into its own suspense component and likely creating three async chunks which would need to be loaded. Seems like a waste of HTTP requests to me. I'd love to brainstorm some better options with you or someone from this team in the future, but I don't think this should block this PR right now.

Can you help me get some time scheduled with the people who setup this plugin this way?

Reply:
Let's get @sethmlarson included in this as well. He'll be back next week, so he should then be able to set up some time to dive into the details with you.

Reply:
Sounds great, thanks! I just saw https://github.com/elastic/kibana/pull/145014 and it bumps the limit up even further. When both of these are merged I expect that the customIntegrations plugin will be downloading a decent amount of unnecessary code on page load. Are there more integrations we plan on adding in the short term?

Reply:
Yes, for each Elasticsearch client, an integration page will be created.

Reply:
Thanks @miriam-eid, I count 13 clients which which means this is definitely trending towards being a problem worth solving ASAP.